pkg/ebpf/tracer.go

```diff
@@ -149,7 +149,7 @@ func NewTracer(config *Config) (*Tracer, error) {
 	}
 
 	var reverseDNS ReverseDNS = nullReverseDNS{}
-	if config.DNSInspection {
+	if shouldEnableSocketFilter(config) {
```

Comment:
You can call `shouldEnableSocketFilter()` once above and use here here + pass it into `SectionsFromConfig()` as a param. 

Also `isRHELOrCentOS` is already called in this function below so perhaps we bring that up as well to reduce the number of calls to `getPlatform()` (which needs to go through a number of things to figure out the platform...)